webcompat/views.py

```diff
@@ -243,6 +246,46 @@ def show_issue(number):
     if session.get('show_thanks'):
         flash(number, 'thanks')
         session.pop('show_thanks')
+    # All of this is just prototype pseudo-working-code
+    # to explain the concept. It needs to be distributed
+    # in definitions with their own set of tests.
+    # The pull request on webcompat-tests is 516.
+    # We query the issues.db which is a kind of progressive cache.
+    # aka we do not have to run an out of process to build the issues.db
+    # It becomes better little by little with what is really used.
+    # It also can be empty and it will not fail.
+    # Benefits of the two worlds.
+    # We could add more stuff to it and slowly have an issue db
+    # that would speed up our full process and make search easy
+    issue = Issues.query.filter_by(issue_number=number).first()
+    # The issue at first request will never exist.
+    # So we are going to request directly GitHub, but only once.
+    if issue is None:
```

Comment:
Big question: how do we handle edits to issue body content on GitHub to prevent serving stale content? Or labels or comments changes/additions/deletion (assuming we care to cache those things here). Webhooks? Cron? Etags (somehow)? Dunno, food for thought.

Reply:
So for now, we do not have to. The thing is really to solve the 403 issue with a door opened for more stuff. And for the title I put it here in the prototype just to show how it could evolve. Title was an obvious choice because of the bookmarking and tabs thing with more information for users. 

Yes Webhooks is one possibility for updating the content of the DB. I would not cache the comments yet. And if we were doing it, it would probably be wise to put them in a separate DB.

Another possibility, if we keep the principle of failsafe, aka the DB can be erased all the time and the system continue to work, we can be creative for populating the DB. For example, it's possible to have a script feeding the DB by just querying all the issues once. out of webcompat.com processes.

So I propose two steps.

1. Create a proper pull request for this (with observers for measuring performances) aka DB.
2. Deploy on staging and test
3. Remove the observers
4. Deploy on production if we are satisfied.

Reply:
@karlcow let's do it.